desktop-widgets/mainwindow.h

```diff
@@ -218,6 +219,8 @@ public
 	struct dive copyPasteDive;
 	struct dive_components what;
 	QList<QAction *> profileToolbarActions;
+	QStringList recentFiles;
+	static constexpr const int numRecentFiles = 4;
```

Comment:
i would go with a macro `#define NUM_RECENT_FILES 4` on top of `mainwindow.cpp`. i *think* i understand the optimization here, but we tend to avoid `C++11`.

Reply:
...the reason being, that 90% of our developers do not understand some of the the new `C++` features.

Reply:
Ok, no problem.
constexpr simply means "guaranteed to be evaluatable at compiletime". It's necessary if you want to size arrays depending on a variable. In constrast to preprocessor directives it is type safe, etc.

Reply:
i think i've tried `constexpr` some years ago to fill a lookup table on compile time and it worked.